### PR TITLE
Refactor parameters

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
A refactoring bu also a bug fix...

**What is the current behavior?** *(You can also link to an open issue here)*
All code for converting powsybl api `LoadflowParameters` to internal parameters (`AcLoadFlowParameters` and `DcLoadFlowParameters`) is in `OpenLoadProvider` which is not the right place.

**What is the new behavior (if this is a feature change)?**
Everything has been moved to `OpenLoadFlowParameters`

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
@annetill It was not my intend for this PR, but I think I fixed an ugly bug. In the security analysis, when phase shifter control is activated in `LoadFlowParameters`, current violations are not detected because current equations are not created in the equation system (so current is always nan)

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java [hunk 1]
```diff
@@ -327,4 +351,178 @@ public static OpenLoadFlowParameters get(LoadFlowParameters parameters) {
         }
         return parametersExt;
     }
+
+    public static void logDc(LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt) {
+        LOGGER.info("Slack bus selection mode: {}", parametersExt.getSlackBusSelectionMode());
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:356]: Maybe you have to add here ` LOGGER.info("Direct current: {}", parameters.isDc());`, no ?

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java [hunk 2]
```diff
@@ -327,4 +351,178 @@ public static OpenLoadFlowParameters get(LoadFlowParameters parameters) {
...
+    public static AcLoadFlowParameters createAcParameters(Network network, LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt,
+                                                          MatrixFactory matrixFactory, Reporter reporter, boolean breakers, boolean forceA1Var) {
+        AcLoadFlowParameters acParameters = createAcParameters(parameters, parametersExt, matrixFactory, reporter, breakers, forceA1Var);
+        if (parameters.isReadSlackBus()) {
+            acParameters.getNetworkParameters().setSlackBusSelector(new NetworkSlackBusSelector(network, acParameters.getNetworkParameters().getSlackBusSelector()));
+        }
+        return acParameters;
+    }
+
+    public static AcLoadFlowParameters createAcParameters(LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt,
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:461]: Why is this class public? Finally only the two previous ones are used in the rest of the code.

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java [hunk 3]
```diff
@@ -327,4 +351,178 @@ public static OpenLoadFlowParameters get(LoadFlowParameters parameters) {
...
+    public static DcLoadFlowParameters createDcParameters(Network network, LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt,
+                                                          MatrixFactory matrixFactory, boolean forcePhaseControlOffAndAddAngle1Var) {
+        var dcParameters = createDcParameters(parameters, parametersExt, matrixFactory, forcePhaseControlOffAndAddAngle1Var);
+        if (parameters.isReadSlackBus()) {
+            dcParameters.getNetworkParameters().setSlackBusSelector(new NetworkSlackBusSelector(network, dcParameters.getNetworkParameters().getSlackBusSelector()));
+        }
+        return dcParameters;
+    }
+
+    public static DcLoadFlowParameters createDcParameters(LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt,
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:494]: Same remark here.

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java (hunk 1 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:357]: Done

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:462]: It is used in another private project

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java (hunk 3 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:495]: It might be used in another private project

**Review by annetill — APPROVED** [review]

**Pull request merged**